File: TestCrc.java
import java.io.*;
import java.util.zip.*;

public class TestCrc {
    public static void main(String[] args) throws Exception {
        File f1 = new File("test.nes");
        try (FileOutputStream fos = new FileOutputStream(f1)) {
            fos.write(new byte[8192]);
        }

        long crcStream;
        try (InputStream input = new FileInputStream(f1)) {
            CRC32 crc = new CRC32();
            byte[] buffer = new byte[8192];
            int length;
            while ((length = input.read(buffer)) >= 0) {
                crc.update(buffer, 0, length);
            }
            crcStream = crc.getValue();
        }
        System.out.println("stream crc: " + crcStream);

        File zipFile = new File("test.zip");
        try (ZipOutputStream zos = new ZipOutputStream(new FileOutputStream(zipFile))) {
            ZipEntry entry = new ZipEntry("test.nes");
            zos.putNextEntry(entry);
            zos.write(new byte[8192]);
            zos.closeEntry();
        }

        try (ZipFile zip = new ZipFile(zipFile)) {
            ZipEntry entry = zip.getEntry("test.nes");
            System.out.println("zip crc: " + entry.getCrc());
        }
    }
}

File: TestCrc.kt
import java.io.File
import java.util.zip.CRC32

fun main() {
    val f1 = File("test.nes")
    f1.writeBytes(ByteArray(8192) { 1 })

    val crc2 = CRC32()
    f1.inputStream().use { input ->
        val buffer = ByteArray(8192)
        var length: Int
        while (input.read(buffer).also { length = it } >= 0) {
            crc2.update(buffer, 0, length)
        }
    }
    println("f1 stream crc: ${crc2.value}")
}
